tools/cli-generator/README.md

```diff
@@ -0,0 +1,45 @@
+# Atlas CLI Generator
+
+:warning: PREVIEW :warning: this tools is still in preview and it might be not suitable for all use cases.
+
+## Usage
+
+Open `spec.yaml` and add to your needs later run `make gen-code`.
+
+### Adding Stores
+
+Add a new entry like this:
+
+```yaml
+  - base_file_name: data_lake_pipeline # this will be internal/store/data_lake_pipeline.go
+    template: store # only template available for stores
+    creator: # possible versions include `creator`, `updater`, `describer`, `lister` and `deleter`
+      name: PipelineCreator # interface name
+      method: CreatePipeline # interface func
+      sdk_method: DataLakePipelinesApi.CreatePipeline # API from SDKv2 to call
+      arg_type: 'atlasv2.IngestionPipeline' # Argument type
+      arg_name: IngestionPipeline # Argument method to call on SDK
+      return_type: '*atlasv2.IngestionPipeline' # Type the SDK returns
+```
+
+### Adding Commands
+
+Add a new entry like this:
+
+```yaml
+  - command_path: atlas dataLake pipeline # parent command path
+    package_name: pipeline # package name for the folder
```

Comment:
can't this be inferred from the path? `atlas dataLake pipeline` -> `/cli/atlas/data_lake/pipeline`?

Reply:
I think many things can be guessed improved here, templates can be inferred, paths, command structure, etc. My goal here was to make something that could get the job done and think about improvements as we go.

My initial version was using openapi as spec to derive things, it proved difficult for a few reasons. Now I think the best course of action is to use the thing as is and list a couple of improvements to be done on our next iteration.

Reply:
agree we can leave those for future improvements but one thing I'm wondering is, how much time are we saving by writing this spec file vs copy/pasting a command and editing the pieces that are different, this is something to keep in mind the process should aim to reduce time and effort at the end